pkg/datapath/linux/probes/probes.go

```diff
@@ -386,10 +388,137 @@ func HaveBoundedLoops() error {
 }
 
 // HaveFibIfindex checks if kernel has d1c362e1dd68 ("bpf: Always return target
-// ifindex in bpf_fib_lookup") which is 5.10+. This got merged in the same kernel
-// as the new redirect helpers.
-func HaveFibIfindex() error {
-	return features.HaveProgramHelper(ebpf.SchedCLS, asm.FnRedirectPeer)
+// ifindex in bpf_fib_lookup") which is 5.10+ (and backports). It does so by
+// checking that ifindex has been updated as a result of the RIB lookup even
+// though the bpf_fib_lookup failed (no L2 resolution for the NH)
+func HaveFibIfindex() (err error) {
+	defer func() {
+		if err != nil && !errors.Is(err, ebpf.ErrNotSupported) {
+			log.WithError(err).Fatal("failed to probe HaveFibIfindex")
+		}
+
+		if errors.Is(err, ebpf.ErrNotSupported) {
+			log.Debug("HaveFibIfindex NOT supported")
+		}
+	}()
+
+	ns, err := netns.New()
+	if err != nil {
+		return fmt.Errorf("create netns: %w", err)
+	}
+	defer ns.Close()
+
+	return ns.Do(func() error {
+		lo, err := netlink.LinkByName("lo")
+		if err != nil {
+			return fmt.Errorf("unable to get loopback iface: %w", err)
+		}
+
+		//Create dummy iface
+		dummy := &netlink.Dummy{
+			LinkAttrs: netlink.LinkAttrs{
+				Name: "dummy",
+			},
+		}
+		if err := netlink.LinkAdd(dummy); err != nil {
+			return fmt.Errorf("unable to create dummy iface: %w", err)
+		}
+
+		//Bring link up on lo&dummy
+		if err := netlink.LinkSetUp(lo); err != nil {
+			return fmt.Errorf("unable to bring up loopback iface: %w", err)
+		}
+		if err := netlink.LinkSetUp(dummy); err != nil {
+			return fmt.Errorf("unable to bring up dummy iface: %w", err)
+		}
+
+		//Addr
+		addrStr := "10.0.0.1/24"
+		addr, _ := netlink.ParseAddr(addrStr)
+		if err := netlink.AddrAdd(dummy, addr); err != nil {
+			return fmt.Errorf("unable to add addr %s to dummy iface: %w", addrStr, err)
+		}
+
+		//Add dummy unresolvable route
+		gwAddrStr := "10.0.0.2"
+		dstCIDR := "1.2.3.4/32"
+		_, dst, _ := net.ParseCIDR(dstCIDR)
+		route := &netlink.Route{
+			LinkIndex: dummy.Attrs().Index,
+			Dst:       dst,
+			Gw:        net.ParseIP(gwAddrStr),
+		}
+		if err := netlink.RouteAdd(route); err != nil {
+			return fmt.Errorf("unable to add route %s via %s dev dummy: %w", dstCIDR, gwAddrStr, err)
+		}
+
+		//netlink stores bytes in NBO, so use Native to conver to uint32
+		nhBe := binary.NativeEndian.Uint32(dst.IP.To4())
+
+		//Attach eBPF program to call bpf_fib_lookup()
+		probe_name := "have_fib_index_probe"
```

Comment:
Looks like this isn't actually put into the ProgramSpec, so maybe it doesn't really need to be a variable. Or it should go into the spec.

Reply:
It's used in traces
https://github.com/cilium/cilium/blob/ab0a12a428cba40e7e662627ceee1c8d5881fd94/pkg/datapath/linux/probes/probes.go#L506

Reply:
Okay, in that case, shouldn we also put it in the ProgramSpec?

Reply:
>Okay, in that case, shouldn we also put it in the ProgramSpec?

Yes, I will do that